bridges/CssSelector2Bridge.php

```diff
@@ -0,0 +1,421 @@
+            }
+        }
+
+        foreach ($entry_elements as $uri => $element) {
+            $entry = $this->parseEntryElement(
+                $element,
+                $title_selector,
+                $author_selector,
+                $category_selector,
+                $time_selector,
+                $content_cleanup,
+                $remove_styling,
+                $this->feedName
+            );
+
+            $entry['uri'] = $uri;
+            $this->items[] = $entry;
+        }
+    }
+
+    /**
+     * Filter a list of URLs using a pattern and limit
+     * @param array $links List of URLs
+     * @param string $url_pattern Pattern to look for in URLs
+     * @param int $limit Optional maximum amount of URLs to return
+     * @return array Array of URLs
+     */
+    protected function filterUrlList($links, $url_pattern, $limit = 0)
+    {
+        if (!empty($url_pattern)) {
+            $url_pattern = '/' . str_replace('/', '\/', $url_pattern) . '/';
+            $links = array_filter($links, function ($url) {
+                return preg_match($url_pattern, $url) === 1;
+            });
+        }
+
+        if ($limit > 0 && count($links) > $limit) {
+            $links = array_slice($links, 0, $limit);
+        }
+
+        return $links;
+    }
+
+    /**
+     * Retrieve title from webpage URL or DOM
+     * @param string|object $page URL or DOM to retrieve title from
+     * @param string $title_cleanup optional string to remove from webpage title, e.g. " | BlogName"
+     * @return string Webpage title
+     */
+    protected function getTitle($page, $title_cleanup)
+    {
+        if (is_string($page)) {
+            $page = getSimpleHTMLDOMCached($page);
+        }
+        $title = html_entity_decode($page->find('title', 0)->plaintext);
+        if (!empty($title)) {
+            $title = trim(str_replace($title_cleanup, '', $title));
+        }
+
+        return $title;
+    }
+
+    /**
+     * Remove all elements from HTML content matching cleanup selector
+     * @param string|object $content HTML content as HTML object or string
+     * @return string|object Cleaned content (same type as input)
+     */
+    protected function cleanArticleContent($content, $cleanup_selector, $remove_styling)
+    {
+        $string_convert = false;
+        if (is_string($content)) {
+            $string_convert = true;
+            $content = str_get_html($content);
+        }
+
+        if (!empty($cleanup_selector)) {
+            foreach ($content->find($cleanup_selector) as $item_to_clean) {
+                $item_to_clean->outertext = '';
+            }
+        }
+
+        if ($remove_styling) {
+            // Get rid of classes
+            $content = preg_replace('/(<[^>]+) class=".*?"/i', '$1', $content);
+            // Get rid of inline styling
+            $content = preg_replace('/(<[^>]+) style=".*?"/i', '$1', $content);
+        }
```

Comment:
At this point, `$content` is an HTML object, not a string, so you should be able to get all elements having a class or style, then remove them. Something like this (untested):
```php
foreach (['class', 'style'] as $attribute_to_remove) {
    foreach ($content->find('[' . $attribute_to_remove . ']') as $item_to_clean) {
        $item_to_clean->removeAttribute($attribute_to_remove);
    }
}
```